macros/src/lib.rs

```diff
@@ -325,8 +332,18 @@ fn generate_impl_block_header(
         G::Lifetime(LifetimeParam { lifetime, .. }) => quote!(#lifetime),
     });
 
-    let where_bound = generate_where_clause(generics, dependencies);
-    quote!(impl <#(#bounds),*> ::ts_rs::TS for #ty <#(#type_args),*> #where_bound)
+    let where_bound = bounds.map_or_else(
+        || {
+            let bounds = generate_where_clause(generics, dependencies);
+            quote! { #bounds }
+        },
+        |bounds| {
+            let bounds = syn::parse_str::<TokenStream>(bounds).expect("malformed `bounds`");
+            quote! { where #bounds }
+        },
+    );
```

Comment:
I think I'd prefer a `match` here. No problem if you prefer map_or_else, though.